src/client/Microsoft.Identity.Client/MsalManagedIdentityException.cs

```diff
@@ -87,7 +87,7 @@ public class MsalManagedIdentityException : MsalServiceException
         /// <summary>
         /// Specifies the managed identity source from which the exception initiates.
         /// </summary>
-        public ManagedIdentitySource ManagedIdentitySource { get; }
+        public new ManagedIdentitySource ManagedIdentitySource { get; }
```

Comment:
The way we did this is a bit funky. `MsalException.ManagedIdentitySource` is a const string key into the AdditionalExceptionData but this is of `ManagedIdentitySource` type...

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/be489b50f7941add10166c454bf109fbd602a1fa/src/client/Microsoft.Identity.Client/MsalException.cs#L54C29-L54C50

Reply:
It's a fair point. `ManagedIdentitySource` is still a public enum btw. But I think our Exception's "AdditionalData" is a string 2 string dictionary, so it kind of makes sense to have a string property here. Do you see a better solution?

Reply:
Both properties are public; so can't change them and we're obsoleting this exception type. So I suppose just keep as is.